Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Login & Signup Flow/NonprofitSignupViewController.m

```diff
@@ -28,20 +29,52 @@ @implementation NonprofitSignupViewController
 - (void)viewDidLoad {
     [super viewDidLoad];
     self.finishedSavingBoth = false;
-    self.nonprofit = [Nonprofit new]; //initWithDict
+    self.nonprofit = [Nonprofit new];
+    
+
     //TODO: figure out document uplaod
     // https://stackoverflow.com/questions/37296929/implement-document-picker-in-swift-ios
 }
 
 - (IBAction)addPictureButtonTapped:(id)sender {
+    UIImagePickerController *imagePickerVC = [UIImagePickerController new];
+    imagePickerVC.delegate = self;
+    imagePickerVC.allowsEditing = YES;
+    BOOL cameraSourceAvail = [UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypeCamera];
+    BOOL photoLibAvail = [UIImagePickerController isSourceTypeAvailable:UIImagePickerControllerSourceTypePhotoLibrary];
+    if (cameraSourceAvail && photoLibAvail) {
+        //UIActionSheet to pick a source type
+        UIAlertController *sourceTypePicker = [UIAlertController alertControllerWithTitle:@"" message:@"" preferredStyle:UIAlertControllerStyleActionSheet];
+        UIAlertAction *camera = [UIAlertAction actionWithTitle:@"Camera" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+            imagePickerVC.sourceType = UIImagePickerControllerSourceTypeCamera;
+        }];
+        UIAlertAction *photoLib = [UIAlertAction actionWithTitle:@"Photo Library" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
+            imagePickerVC.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+        }];
+        [sourceTypePicker addAction:camera];
+        [sourceTypePicker addAction:photoLib];
+        [self presentViewController:sourceTypePicker animated:YES completion:nil];
+    } else {
+        imagePickerVC.sourceType = UIImagePickerControllerSourceTypePhotoLibrary;
+    }
+    [self presentViewController:imagePickerVC animated:YES completion:nil];
+}
+
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info {
+    
+    UIImage *originalImage = info[UIImagePickerControllerOriginalImage];
```

Comment:
Please delete any unused variables. 
Is the edited image always the same as the original image when the user doesn't edit anything?

Reply:
Thanks, good catch. Yes, should be!